File: sort/shell_sort/cpp/shell sort.cpp
#include<iostream>
using namespace std;
int main()
{
	int a[20],n,j,gap,item;
	cout<<"Enter the number of elements: ";
	cin>>n;
	gap=n/2;
	for(int i=0;i<n;i++)
		cin>>a[i];
	while(gap>=1)
	{
		for(int i=gap;i<n;i=i+gap)
		{
			item=a[i];
			for(j=i-gap;j>=0 && a[j]>item;j=j-gap)
			{
				a[j+gap]=a[j];
			}
			a[j+gap]=item;
		}
		gap=gap/2;
	}
	for(int i=0;i<n;i++)
		cout<<a[i]<<" ";
	return 0;
}

File: sort/shell_sort/java/ShellSort.java
import java.util.*;

public class ShellSort {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    int[] elements;
    System.out.println("Enter the number of elements: ");
    int j, n = sc.nextInt();
    elements = new int[n];
    int gap = n / 2;
    System.out.println("Enter elements: ");
    for(int i = 0; i < n; i++)
      elements[i] = sc.nextInt();
    while(gap >= 1) {
      for(int i = gap; i < n; i += gap) {
        int item = elements[i];
        for(j = i - gap; j >= 0 && elements[j] > item; j -= gap) {
          elements[j + gap] = elements[j];
        }
        elements[j + gap] = item;
      }
      gap /= 2;
    }
    for(int i = 0; i < n; i++) {
      System.out.print(elements[i] + " ");
    }
    System.out.println();
  }
}
